### PR TITLE
[MXNET-85] Update the OSX Installation guide for Scala Users

### PR DESCRIPTION
**Pull request by lanking520**
## Description ##
Modification on the setup guide for OSX users @Roshrini @nswamy 

### Essentials ###
- [X] Passed code style checking (`make lint`)
- [X] Changes are complete (i.e. I finished coding on this PR)
- [X] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [X] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- [X] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
Add several changes in the setup document in OSX setup.
Add note for issues in jupyter installation.
Add maven installation as a scala installation requirement.

## Comments ##
N/A

### CODE REVIEW

**Comment by Roshrini** [comment]
@aaronmarkham

File: docs/install/osx_setup.md [hunk 1]
```diff
@@ -65,14 +65,19 @@ Install the dependencies, required for MXNet, with the following commands:
...
+	# Get pip
+	easy_install pip
 	# For visualization of network graphs
 	pip install graphviz
 	# Jupyter notebook
 	pip install jupyter
 ```
 
+#### Note
+If you already have Python installed on your Mac, you may face the issue while installing jupyter caused by restricted access in System(see [here](https://stackoverflow.com/questions/33004708/osx-el-capitan-sudo-pip-install-oserror-errno-1-operation-not-permitted)). In that case, use brew resintall python to install another version of Python under homebrew. Use pip2 instead of pip to install the following packages. It also requires you to run a python program using python2 (e.g $ python2 example.py) instead of python.
```
Comment (Roshrini) [inline docs/install/osx_setup.md]: Typo in: use "brew **resintall** python" to install another version of Python under homebrew.

Comment (lanking520) [reply docs/install/osx_setup.md]: Done! Make another commit to fix that issue.

**Comment by CodingCat** [comment]
Hi, the community has passed to vote about associating the code changes with JIRA (https://lists.apache.org/thread.html/ab22cf0e35f1bce2c3bf3bec2bc5b85a9583a3fe7fd56ba1bbade55f@%3Cdev.mxnet.apache.org%3E)

We have updated the guidelines for contributors in https://cwiki.apache.org/confluence/display/MXNET/Development+Process, please ensure that you have created a JIRA at https://issues.apache.org/jira/projects/MXNET/issues/ to describe your work in this pull request and include the JIRA title in your PR as [MXNET-xxxx] your title where MXNET-xxxx is the JIRA id

Thanks!

**Comment by Roshrini** [comment]
@CodingCat Do we need to create Jira for document changes as well?

**Pull request closed without merging**